gems/pending/postgres_ha_admin/failover_databases.rb

```diff
@@ -1,80 +1,59 @@
-require 'postgres_ha_admin/postgres_ha_admin'
 require 'pg'
 require 'pg/dsn_parser'
 
 module PostgresHaAdmin
   class FailoverDatabases
     TABLE_NAME = "repl_nodes".freeze
 
-    attr_reader :yml_file, :connection_hash, :logger, :log_file
+    attr_reader :yml_file, :logger
 
-    def initialize(config_dir, log_dir, connection_params_hash)
-      @yml_file = Pathname.new(config_dir).join(DB_YML_FILE)
-      @log_file = Pathname.new(log_dir).join(LOG_FILE_NAME)
-      @logger = Logger.new(@log_file)
-      @logger.level = Logger::INFO
-      @connection_hash = connection_params_hash
+    def initialize(yml_file, logger)
+      @yml_file = yml_file
+      @logger = logger
     end
 
-    def refresh_databases_list(connection = nil)
-      query_repmgr(connection)
+    def active_databases
+      all_databases.select { |record| record[:active] }
     end
 
-    def all_databases(connection = nil)
-      if File.exist?(yml_file)
-        begin
-          YAML.load_file(yml_file)
-        rescue IOError => err
-          logger.error("#{err.class}: #{err}")
-          logger.error(err.backtrace.join("\n"))
-          []
-        end
-      else
-        query_repmgr(connection)
-      end
+    def update_failover_yml(connection)
+      arr_yml = query_repmgr(connection)
+      File.write(yml_file, arr_yml.to_yaml) if arr_yml.size > 0
+    rescue IOError => err
+      logger.error("#{err.class}: #{err}")
+      logger.error(err.backtrace.join("\n"))
     end
 
-    def standby_databases(connection = nil)
-      all_databases(connection).select { |record| record[:type] == 'standby' }
-    end
+    private
 
-    def active_standby_databases(connection = nil)
-      all_databases(connection).select { |record| record[:type] == 'standby' && record[:active] == true }
+    def all_databases
+      YAML.load_file(yml_file)
+    rescue IOError => err
```

Comment:
Instead of rescuing here you can probably just `return [] unless File.exist(yml_file)` at the top then do the load.